File: WaTF-Bank-iOS/watf-bank/PageController/Summary.m
#import "Util.h"
#import "Http.h"

@interface Summary:UIViewController
@property (weak, nonatomic) IBOutlet UILabel *accountNo;
@property (weak, nonatomic) IBOutlet UILabel *balance;
@property (weak, nonatomic) IBOutlet UILabel *tel;
@property (weak, nonatomic) IBOutlet UILabel *username;
@end

@implementation Summary
- (void)viewDidLoad
{
    [super viewDidLoad];
    NSDictionary *input = [NSDictionary dictionaryWithObjectsAndKeys:[Util getKeychain:@"accountNo"],@"accountNo",[Util getKeychain:@"token"],@"token", nil];
    [Http post:@"accountSummary" :input :^(NSDictionary *res)
     {
         if(res==nil)
         {
             [Util alert:self :@"Connection Problem"];
             return;
         }
         if([[res[@"message"] lowercaseString] isEqualToString:@"success"])
         {
             self.username.text = [NSString stringWithFormat:@"%@ : %@",@"Username",res[@"username"]];
             self.tel.text = [NSString stringWithFormat:@"%@ : %@",@"Telephone",res[@"tel"]];
             self.balance.text = [NSString stringWithFormat:@"%@ : %@",@"Balance",res[@"balance"]];
             self.accountNo.text = [NSString stringWithFormat:@"%@ : %@",@"Account No",[Util getKeychain:@"accountNo"]];
         }
         else
         {
             [Util alert:self :@"An Error Occurred"];
         }
     }];
}
@end

File: iOS-Swift/watf-bank/PageController/Summary.swift
import UIKit

class Summary : UIViewController {
    
    @IBOutlet weak var username: UILabel!
    @IBOutlet weak var tel: UILabel!
    @IBOutlet weak var balance: UILabel!
    @IBOutlet weak var accountNo: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        guard let token = KeyChain.load("token"), let acc = KeyChain.load("accountNo") else {
            Util.alert(self, "Invalid Token")
            return
        }
        let input = ["accountNo":acc,"token":token]
        Http().post(input, "accountSummary", completionHandler: {(re: [String:Any]) in
            var res = re as! [String:String]
            if let e = res["error"] {
                Util.alert(self, e)
            }
            else if res["message"] == "Success" {
                self.username.text = "Username : "+res["username"]!
                self.tel.text = "Telephone : "+res["tel"]!
                self.balance.text = "Balance : "+res["balance"]!
                self.accountNo.text = "Account No : "+acc
            }
            else if let m = res["message"] {
                Util.alert(self, m)
            }
            else {
                Util.alert(self, "An Error Occurred")
            }
        })
    }
}
